.eslintrc.js

```diff
@@ -41,6 +41,15 @@ module.exports = {
   'rules': Object.assign({},
     todo,
     {
+      // turned off otherwise non-admin modules will complain about importing components from admin
+      // via the novel silverstripe js component sharing setup
+      'import/no-extraneous-dependencies': [
+        'off'
+      ],
+      // turned off to allow the using of the js file extension for /boot/index.js
+      'import/extensions': [
```

Comment:
Couldn't we just remove the file extension in those imports? I would rather stay as close as possible to the base `eslint-config-airbnb` rule set.

Reply:
Means we'd need to update all the affected modules individually e.g. https://github.com/silverstripe/silverstripe-campaign-admin/blob/1/client/src/bundles/bundle.js and however many else

We'd also need to do this for older versions too such as 1.9.  Massive pain :-/

Reply:
There's a large number of files affected, doing a regex search `require\(.+?\.js'` on a sink install netted 47 results in 21 files

A few random examples:
- https://github.com/silverstripe/silverstripe-admin/blob/1/client/src/legacy/SecurityAdmin.js#L6
- https://github.com/silverstripe/silverstripe-ckan-registry/blob/1/client/src/bundles/bundle.js#L2
- https://github.com/silverstripe/silverstripe-admin/blob/1/client/src/legacy/ssui.core.js#L3

I really don't think all the pull-requests required to fix this justifies the trivial linting benefit

Reply:
If every time we get linting errors, we disable our linting rules, we might as well not have any linting rules at all. As far I'm concern, having a rule to disallow adding useless `.js` extension in require statements is perfectly sensible.

If we don't have the time to fix these linting errors right now, lets change the constraint in `package.json` so we get the older version of `eslint-config-airbnb` that allow `require` statements. But we'll have to get to it eventually.

Reply:
Agree with Max 👍